WireDomain/Sources/WireDomain/Repositories/Conversations/ConversationLocalStore+MLS.swift

```diff
@@ -19,11 +19,683 @@
 import WireAPI
 import WireDataModel
 
-/// An extension that encapsulates storage operations related to conversation MLS.
+/// An extension that encapsulates storage operations related to MLS.
 
 extension ConversationLocalStore {
+    
+    // MARK: - Public
+    
+    public func addMLSMessage(
+        _ encryptedMessage: String,
+        mlsGroupID: MLSGroupID,
+        mlsConversation: ZMConversation,
+        senderID: UUID,
+        senderDomain: String,
+        subconversation: String?,
+        date: Date?
+    ) async {
+        
+        let decryptionResults = await decryptMLSMessage(
+            message: encryptedMessage,
+            mlsGroupID: mlsGroupID,
+            subconversation: subconversation
+        )
+        
+        await processMLSMessageDecryptionResults(
+            decryptionResults,
+            mlsConversation: mlsConversation,
+            senderID: senderID,
+            senderDomain: senderDomain,
+            date: date
+        )
+        
```

Comment:
In legacy code, we were handling MLS message decryption and actual message processing in two different places : 

`EventDecoder+MLS` would first decode the MLS message add event payload and decrypt the MLS message. 

Then, new events with the decrypted message would be manually created and propagated so the `ClientMessageRequestStrategy` would consume them.